azureeyemodule/app/iot/iot_interface.cpp

```diff
@@ -147,47 +147,66 @@ static inline std::unordered_map<std::string, std::string> parse_onvif_message(c
     // This naive format works for something as simple as this for now, but we should really consider
     // using a better serialization/deserialization mechanism.
 
-    const std::string delimiters = "-_";
+    const std::string delimiter1 = "-"; // the delimiter to seperate different feature
+    const std::string delimiter2 = "_"; // the delimiter to deperate different parameters for each feature
     std::string dup = std::string(msg);
 
     // Tokenize the string into a vector of strings
     std::vector<std::string> tokens_and_values;
     size_t pos = 0;
-    while ((pos = dup.find_first_of(delimiters)) != std::string::npos)
+    while ((pos = dup.find_first_of(delimiter1)) != std::string::npos)
     {
         std::string part = dup.substr(0, pos);
+	printf("- parsed : %s \n", part.c_str());
         if (part.length() > 0)
         {
-            tokens_and_values.push_back(part);
+            tokens_and_values.push_back(part+"_");
```

Comment:
Since you are already making changes to this file, could you please change this to include spaces around the "+" sign? It's more in line with the style of the rest of the code.